docs/resources/client/models/items.md

```diff
@@ -0,0 +1,1113 @@
+    }
+
+    // Render the model
+    @Override
+    public void render(@Nullable Boolean barVisible, ItemDisplayContext displayContext, PoseStack pose, MultiBufferSource bufferSource, int light, int overlay, boolean hasFoil) {
+        this.model.renderToBuffer(pose, this.material.buffer(bufferSource, barVisible ? RenderType::entityCutout : RenderType::entitySolid), light, overlay);
+    }
+}
+```
+
+Next is the `SpecialModelRenderer.Unbaked` instance. This should contain data that can be read from a file to determine what to pass into the special renderer. This also contains two methods: `bake`, which is used to construct the special renderer instance; and `type`, which defines the `MapCodec` to use for encoding/decoding to file.
+
+```java
+public record ExampleSpecialRenderer(Model model, Material material) implements SpecialModelRenderer<Boolean> {
+
+    // ...
+
+    public record Unbaked(ResourceLocation texture) implements SpecialModelRenderer.Unbaked {
+
+        public static final MapCodec<ExampleSpecialRenderer.Unbaked> MAP_CODEC = ResourceLocation.CODEC.fieldOf("texture")
+            .xmap(ExampleSpecialRenderer.Unbaked::new, ExampleSpecialRenderer.Unbaked::texture);
+
+        @Override
+        public MapCodec<ExampleSpecialRenderer.Unbaked> type() {
+            return MAP_CODEC;
+        }
+
+        @Override
+        public SpecialModelRenderer<?> bake(EntityModelSet modelSet) {
+            // Get the model and the material to render
+            return new ExampleSpecialRenderer(...);
+        }
+    }
+}
+```
+
+Finally, we register the objects to their necessary locations. For the client items, this is done via `RegisterSpecialModelRendererEvent` on the [mod event bus][modbus]. If the special renderer should also be used as part of a `BlockEntityRenderer`, such as when rendering in some item-like context (e.g., enderman holding the block), then an `Unbaked` version for the block should be registered via `RegisterSpecialBlockModelRendererEvent` on the [mod event bus][modbus].
+
+```java
+// In some class where the event is registered to the mod event bus
+@SubscribeEvent
+public static void registerSpecialRenderers(RegisterSpecialModelRendererEvent event) {
+    event.register(
+        // The name to reference as the type
+        ResourceLocation.fromNamespaceAndPath("examplemod", "example_special"),
+        // The map codec
+        ExampleSpecialRenderer.Unbaked.MAP_CODEC
+    )
+}
+
+// For rendering a block in an item-like context
+// Assume some DeferredBlock<ExampleBlock> EXAMPLE_BLOCK
+@SubscribeEvent
+public static void registerSpecialBlockRenderers(RegisterSpecialBlockModelRendererEvent event) {
+    event.register(
+        // The block to render for
+        EXAMPLE_BLOCK.get()
+        // The unbaked instance to use
+        new ExampleSpecialRenderer.Unbaked(ResourceLocation.fromNamespaceAndPath("examplemod", "entity/example_special"))
+    )
+}
+```
+
+## Rendering an Item
```

Comment:
Is this entire section relevant to modders, for the sake of being able to hook into it somehow?

Reply:
Ah, then this should probably be stated more clearly.

Reply:
Yep, I tweaked the section title and updated the first sentence to reflect the intended usecase.